lib/active_merchant/billing/gateways/elavon.rb

```diff
@@ -56,27 +56,32 @@ def purchase(money, payment_method, options = {})
           add_customer_email(xml, options)
           add_test_mode(xml, options)
           add_ip(xml, options)
-          add_auth_purchase_params(xml, options)
+          add_auth_purchase_params(xml, payment_method, options)
           add_level_3_fields(xml, options) if options[:level_3_data]
         end
         commit(request)
       end
 
-      def authorize(money, creditcard, options = {})
+      def authorize(money, payment_method, options = {})
         request = build_xml_request do |xml|
           xml.ssl_vendor_id         @options[:ssl_vendor_id] || options[:ssl_vendor_id]
           xml.ssl_transaction_type  self.actions[:authorize]
           xml.ssl_amount            amount(money)
 
-          add_salestax(xml, options)
+          if payment_method.is_a?(String)
+            add_token(xml, payment_method)
+          else
+            add_creditcard(xml, payment_method) unless options[:ssl_token]
+          end
```

Comment:
Adds support for Authorize transctions with a token. I'm not sure why this wasn't added originally.